Src/EB/AMReX_EB_chkpt_file.cpp

```diff
@@ -0,0 +1,371 @@
+    AMREX_ASSERT(nlevs == 1);
+
+    std::getline(is, line);
+    {
+        std::istringstream lis(line);
+        int i = 0;
+        while (lis >> word) {
+            prob_lo[i++] = std::stod(word);
+        }
+    }
+
+    std::getline(is, line);
+    {
+        std::istringstream lis(line);
+        int i = 0;
+        while (lis >> word) {
+            prob_hi[i++] = std::stod(word);
+        }
+    }
+
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(Math::abs(prob_lo[idim] - geom.ProbLo()[idim]) < std::numeric_limits<Real>::epsilon(),
+                                         "EB2::ChkptFile cannot read from a different problem domain");
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(Math::abs(prob_hi[idim] - geom.ProbHi()[idim]) < std::numeric_limits<Real>::epsilon(),
+                                         "EB2::ChkptFile cannot read from a different problem domain");
+    }
+
+    IntVect ngrow_chkptfile;
+    std::getline(is, line);
+    {
+        std::istringstream lis(line);
+        int i = 0;
+        while (lis >> word) {
+            ngrow_chkptfile[i++] = std::stoi(word);
+        }
+    }
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(ngrow_chkptfile == ngrow_finest, "EB2::ChkptFile cannot read from different ngrow");
+
+    bool edf_chkptfile;
+    is >> edf_chkptfile;
+    gotoNextLine(is);
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(extend_domain_face == edf_chkptfile,
+                                     "EB2::ChkptFile cannot read from different extend_domain_face");
+
+    int mgs_chkptfile;
+    is >> mgs_chkptfile;
+    gotoNextLine(is);
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(max_grid_size == mgs_chkptfile,
+                                     "EB2::ChkptFile cannot read from different max_grid_size");
+
+    BoxArray cut_grids_ba, covered_grids_ba;
+
+    Print() << "Loading cut_grids\n";
+    cut_grids_ba.readFrom(is);
+    gotoNextLine(is);
+
+    if (is.peek() != EOF) {
+        Print() << "Loading covered_grids\n";
+        covered_grids_ba.readFrom(is);
+        gotoNextLine(is);
+    }
+
+    BoxList cut_bl;
+    for (int nb = 0; nb < cut_grids_ba.size(); nb++) {
+        Box b(cut_grids_ba[nb]);
+        cut_bl.push_back(b);
+    }
+
+    BoxList covered_bl;
+    for (int nb = 0; nb < covered_grids_ba.size(); nb++) {
+        Box b(covered_grids_ba[nb]);
+        covered_bl.push_back(b);
+    }
+
+    cut_grids.define(cut_bl);
+    dmap.define(cut_grids, ParallelDescriptor::NProcs());
+
+    covered_grids.define(covered_bl);
+
+    // volfrac
+    {
+        Print() << "  Loading " << m_volfrac_name << std::endl;
+
+        volfrac.define(cut_grids, dmap, 1, ng_gfab);
+        volfrac.setVal(1.); // regular value
+
+        auto prefix = MultiFabFileFullPrefix(0, m_restart_file, level_prefix, m_volfrac_name);
+        MultiFab mf(The_Pinned_Arena());
+        VisMF::Read(mf, prefix);
+        volfrac.ParallelCopy(mf, 0, 0, 1, 0, ng_gfab, geom.periodicity());
```

Comment:
```suggestion
        volfrac.define(cut_grids, dmap, 1, ng_gfab);
        auto prefix = MultiFabFileFullPrefix(0, m_restart_file, level_prefix, m_volfrac_name);
        VisMF::Read(volfrac, prefix);
```
I think we can read the data directly into volfrac.  VisMF::Read works for non-managed GPU memory too.